include/dlaf/util_matrix.h

```diff
@@ -40,5 +40,31 @@ void assert_blocksize_square(const Matrix& matrix, std::string function, std::st
                                 " is not square.");
 }
 
+/// @brief Verify if dlaf::Matrix is on local memory
+///
+/// @tparam Matrix refers to a dlaf::Matrix object
+/// @throws std::invalid_argument if the matrix is not stored on local memory
+template <class Matrix>
+void assert_local_matrix(const Matrix& matrix, std::string function, std::string mat_name) {
+  assert(matrix.size().isValid());
+  if (matrix.distribution().commGridSize() != comm::Size2D{1, 1})
+    throw std::invalid_argument(function + ": " + "Matrix " + mat_name + " is not local.");
+}
+
+/// @brief Verify that distributed dlaf::Matrix correspond to the one in the Communicator
```

Comment:
typo: *corresponds*

However, I would try to be more specific. It is not clear what `correspond` means, I would give more details, e.g. "their sizes and rank ids must match".